### ISSUE TITLE
Annotations allowedTargets is ignored

### ISSUE THREAD

**Issue by postatum** [aml-org/amf#492]
### Issue
Running code below should fail or report validation error because according to RAML 1.0 spec provided RAML input is invalid.

The spec says:
> The locations to which annotations are restricted. If this node is specified, annotations of this type may be applied only on a node corresponding to one of the locations. The value MUST be one or more of the options described in the Target Locations.

Spec link:
https://github.com/raml-org/raml-spec/blob/master/versions/raml-10/raml-10.md/#declaring-annotation-types

### AMF Version
4.0.0-SNAPSHOT

### Code
```sh
$ java -jar amf-4.0.0-SNAPSHOT.jar validate -in "RAML 1.0" -mime-in "application/yaml" -p "RAML" file://api.raml
{
  "@type": "http://www.w3.org/ns/shacl#ValidationReport",
  "http://www.w3.org/ns/shacl#conforms": true
}
```

### api.raml
Is true for all the values of allowedTargets.
```raml
#%RAML 1.0
title: Illustrating allowed targets
mediaType: application/json
annotationTypes:
  test:
    allowedTargets: API
  test2:
    (test): hi

```

**Comment by BlockLucas** [comment]
https://www.mulesoft.org/jira/browse/APIMF-1523

**Comment by gutee** [comment]
This is fixed and stated for 3.3.0 release
